src/main/java/org/icgc/argo/program_service/services/ProgramService.java

```diff
@@ -106,7 +107,31 @@ public ProgramService (
   }
 
   public Optional<ProgramEntity> getProgram(@NonNull UUID uuid) {
-    return programRepository.findById(uuid);
+    log.error("Looking for program with uuid" + uuid);
```

Comment:
We can try using `CancerSpecification` and `PrimarySiteSpecification` to get cancerList and primarySiteList. 
```
public class PrimarySiteSpecification {

  public static Specification<PrimarySiteEntity> containsProgram(@NonNull UUID programId) {
    return (root, query, builder) -> {
      query.distinct(true);
      Join<PrimarySiteEntity, ProgramPrimarySite> psJoin = root.join("programPrimarySites");
      Join<ProgramPrimarySite, ProgramEntity> programJoin = psJoin.join("program");
      return builder.equal(programJoin.<Integer>get(SqlFields.ID), programId);
    };
  }

}
```
then in `ProgramService`, 
```    val primarySiteList = primarySiteRepository.findAll(PrimarySiteSpecification.containsProgram(uuid));```
Simliar for `CancerSpecification`.